src/server_manager/web_app/shadowbox_server.ts

```diff
@@ -177,11 +178,10 @@ export class ShadowboxServer implements server.Server {
           .then(
               (response) => {
                 if (!response.ok) {
-                  const msg = `API request to ${path} failed with status ${response.status}, type ${
-                      response.type}`;
+                  const msg = `API request to ${path} failed with status ${response.status}`;
                   console.error(msg);
                   SentryErrorReporter.logError(msg);
-                  throw new Error(msg);
+                  throw new errors.ServerApiError(msg, response);
```

Comment:
Shouldn't you do some wrapping in the `catch` instead? you may miss some errors. For example, the `JSON.parse` may throw.